app/models/piece.rb

```diff
@@ -10,4 +10,78 @@ class Piece < ActiveRecord::Base
   def user
     color == "white" ? game.white_player : game.black_player
   end
+
+  protected
+  # This method only checks for obstructions along horizontal, vertical, and diagonal moves. Before calling it, 
+  # please make sure that the move provided falls into one of these categories and that an obstruction check is 
+  # actually needed (e.g. the piece isn't a knight, the move is not malformed).
+  def is_move_obstructed?(new_rank, new_file)
+    current_pieces = game.pieces
+
+    # These are assumed to be diagonal moves along a slope of 1
+    if new_rank > rank && new_file > file
+      (1..new_rank - rank).each do |offset|
+        piece_at_offset = current_pieces.select{|piece| piece.rank == rank + offset && piece.file == file + offset}
```

Comment:
Might make sense to have a `get_piece_at(rank, file)` method to reduce some of this code repetition.